### PR TITLE
resolves 1895: port to monterey

### LINKED ISSUE #1895: Port to Monterey
There are several issues:

- cln official releases are not supported on M1, we need to use the git repo see https://github.com/feelpp/cln
- many cmake warnings need to be fixed
- petsc 3.18 has a deprecated macro

this should be part of v110

- [ ] #1986
- [ ] #1987

### PR DESCRIPTION
**Pull request by prudhomm**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [x] Have you successfully run the Feel++ testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----
- closes #1895

### CODE REVIEW

File: feelpp/contrib/CMakeLists.txt [hunk 1]
```diff
@@ -128,21 +128,52 @@ if ( FEELPP_HAS_GLOG )
   set(FEELPP_ENABLED_OPTIONS_CONTRIB "${FEELPP_ENABLED_OPTIONS_CONTRIB} GLog/Contrib")
 endif()
 
+# # cln
+message(STATUS "------------------------------------------------------------")
+message(STATUS "Configuring CLN")
+option(FEELPP_USE_EXTERNAL_CLN "USE_EXTERNAL_FMT" OFF)
```
Comment (vincentchabannes) [inline feelpp/contrib/CMakeLists.txt:134]: FMT -> CLN

File: feelpp/contrib/ginac/ginac/CMakeLists.txt [hunk 2]
```diff
@@ -154,10 +154,10 @@ add_definitions(-DLIBEXECDIR="${LIBEXECDIR}/")
 set_target_properties(feelpp_ginac PROPERTIES
 	SOVERSION ${ginaclib_soversion}
 	VERSION ${ginaclib_version})
-target_link_libraries(feelpp_ginac ${CLN_LIBRARIES} ${CMAKE_DL_LIBS} ${Boost_MPI_LIBRARY} ${Boost_SERIALIZATION_LIBRARY} ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY} ${MPI_LIBRARIES})
+target_link_libraries(feelpp_ginac cln::cln ${CMAKE_DL_LIBS} ${Boost_MPI_LIBRARY} ${Boost_SERIALIZATION_LIBRARY} ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY} ${MPI_LIBRARIES})
```
Comment (vincentchabannes) [inline feelpp/contrib/ginac/ginac/CMakeLists.txt:157]: cln::cln exist with external dependency? (in FindCLN)

File: feelpp/cmake/modules/feelpp.dependencies.cmake [hunk 3]
```diff
@@ -1315,7 +1314,7 @@ endif(FEELPP_ENABLE_GSL)
 option( FEELPP_ENABLE_GMSH "Enable Gmsh Support" ${FEELPP_ENABLE_PACKAGE_DEFAULT_OPTION} )
 if( FEELPP_ENABLE_GMSH )
   if(FEELPP_USE_GMSH_PACKAGE)
-	FIND_PACKAGE(Gmsh)
+	FIND_PACKAGE(GMSH)
```
Comment (vincentchabannes) [inline feelpp/cmake/modules/feelpp.dependencies.cmake:1317]: maybe look in both syntax.

**Review by vincentchabannes — APPROVED** [review]

File: feelpp/cmake/modules/feelpp.dependencies.cmake (hunk 3 above)
Comment (prudhomm) [reply feelpp/cmake/modules/feelpp.dependencies.cmake:1317]: done

File: feelpp/contrib/CMakeLists.txt (hunk 1 above)
Comment (prudhomm) [reply feelpp/contrib/CMakeLists.txt:134]: done

File: feelpp/contrib/ginac/ginac/CMakeLists.txt (hunk 2 above)
Comment (prudhomm) [reply feelpp/contrib/ginac/ginac/CMakeLists.txt:157]: yes this is from cln

Comment (prudhomm) [reply feelpp/contrib/ginac/ginac/CMakeLists.txt:157]: I had to retrieve cln to fix the arm code generation

Comment (prudhomm) [reply feelpp/contrib/ginac/ginac/CMakeLists.txt:157]: an alternative is to compile from source cln and use the recent git version

**Pull request merged**